runbot/models/build_error.py

```diff
@@ -1,22 +1,62 @@
 # -*- coding: utf-8 -*-
+import datetime
 import hashlib
 import json
 import logging
 import re
 
 from collections import defaultdict
 from dateutil.relativedelta import relativedelta
+from dateutil import rrule
 from markupsafe import Markup
 from werkzeug.urls import url_join
 from odoo import models, fields, api
 from odoo.exceptions import ValidationError, UserError
 from odoo.tools import SQL
 
+from lxml import etree
+from lxml.builder import E
+
 from ..fields import JsonDictField
 
 _logger = logging.getLogger(__name__)
 
 
+def get_color(value: int):
+    if value >= 10:
+        return 'red'
+    elif value >= 5:
+        return 'yellow'
+    return 'green'
+
+def draw_svg(values: list[int], max_value: int = 10, get_color_callback = get_color, height: int = 30):
+    return etree.tostring(
+        E.div(
+            {
+                'style': f'height: {height}px',
+            },
+            E.svg(
+                {
+                    'xmlns': "https://www.w3.org/2000/svg",
+                    'viewBox': f"0 0 {len(values)*10} {max_value}",
+                    'style': "border: 1px solid black; height: 100%; width: 100%",
+                    'preserveAspectRatio': 'none',
+                    'shape-rendering': "crispEdges",
+                },
+                *(
+                    E.rect(
+                        fill=get_color(value),
+                        width='9',
+                        height=f'{min(value, max_value)}',
+                        x=f'{idx*10}',
```

Comment:
This changes the padding for uniform distance between rectangles.
```suggestion
                        x=f'{idx*10 + 0.5}',
```